### ISSUE TITLE
Semicolon added on each save

### ISSUE THREAD

**Issue by ripperdoc** [prettier/prettier-vscode#2507]
### Summary

_A certain combination of code and modifications to that code will cause a semicolon to be added on every save, causing syntax error_.

This issue was first raised on vscode [here](https://github.com/microsoft/vscode/issues/146910) but re-directed here. VS Code system info can be found there.

### Github Repository to Reproduce Issue

[Link to a Github repo that can be used to reproduce the issue.
](https://github.com/ripperdoc/repro-vscode-prettier-semicolon)

### Steps To Reproduce:

1. Load the repo with VS Code and VS Code Prettier extension.
2. Ensure format on save is on, that Prettier is chosen as formatter and that the format on save mode is set to either `"editor.formatOnSaveMode": "modificationsIfAvailable"` or `"editor.formatOnSaveMode": "modifications"`.
3. Open the file test_case.js (which is same as before_change.js). Copy the contents of the after_change.js onto the test_case.js file.
4. Save the file and see that a semicolon is added on last line every time you save, causing Syntax error.

### Expected result

No invalid semicolons added.

### Additional information

This only happens on the two mentioned formatOnSaveMode modes. If the format on save is changed to `"editor.formatOnSaveMode": "file"` the error doesn't appear. So it seems to be an interplay between the way VS Code formats only the diffing parts (the diff is compared with what's in the Git repo, see Test commit in the reproduction repo). The error doesn't appear if there is no Git history to diff to.

Extension bisect was used to find that this error appears only with Prettier.

**VS Code Version**: Code 1.66.0 (https://github.com/microsoft/vscode/commit/e18005f0f1b33c29e81d732535d8c0e47cafb0b5, 2022-03-30T05:50:04.419

**Prettier Extension Version**: 9.5.0

**OS and version**: Darwin x64 18.7.0

## Prettier Log Output

```
["INFO" - 4:05:20 PM] Extension Name: esbenp.prettier-vscode.
["INFO" - 4:05:20 PM] Extension Version: 9.5.0.
["INFO" - 4:05:36 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:36 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:36 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:36 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:36 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 130,
  "rangeStart": 39
}
["INFO" - 4:05:36 PM] Formatting completed in 0.06ms.
["INFO" - 4:05:36 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:36 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:36 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:36 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:36 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 313,
  "rangeStart": 313
}
["INFO" - 4:05:36 PM] Formatting completed in 0.007ms.
["INFO" - 4:05:36 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:36 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:36 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:36 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:36 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 611,
  "rangeStart": 501
}
["INFO" - 4:05:36 PM] Formatting completed in 0.028ms.
["INFO" - 4:05:36 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:36 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:36 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:36 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:36 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 313,
  "rangeStart": 39
}
["INFO" - 4:05:36 PM] Formatting completed in 0.008ms.
["INFO" - 4:05:39 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:39 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:39 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:39 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:39 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 130,
  "rangeStart": 39
}
["INFO" - 4:05:39 PM] Formatting completed in 0.009ms.
["INFO" - 4:05:39 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:39 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:39 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:39 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:39 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 313,
  "rangeStart": 313
}
["INFO" - 4:05:39 PM] Formatting completed in 0.006ms.
["INFO" - 4:05:39 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:39 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:39 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:39 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:39 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 611,
  "rangeStart": 501
}
["INFO" - 4:05:39 PM] Formatting completed in 0.012ms.
["INFO" - 4:05:39 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:39 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:39 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:39 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:39 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 795,
  "rangeStart": 792
}
["INFO" - 4:05:39 PM] Formatting completed in 0.008ms.
["INFO" - 4:05:39 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:39 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:39 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:39 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:39 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 313,
  "rangeStart": 39
}
["INFO" - 4:05:39 PM] Formatting completed in 0.008ms.
["INFO" - 4:05:39 PM] Formatting file:///Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js
["INFO" - 4:05:39 PM] Using ignore file (if present) at /Users/martin/dev/src/repro-vscode-prettier-semicolon/.prettierignore
["INFO" - 4:05:39 PM] File Info:
{
  "ignored": false,
  "inferredParser": "babel"
}
["INFO" - 4:05:39 PM] No local configuration (i.e. .prettierrc or .editorconfig) detected, falling back to VS Code configuration
["INFO" - 4:05:39 PM] Prettier Options:
{
  "arrowParens": "always",
  "bracketSpacing": true,
  "endOfLine": "lf",
  "htmlWhitespaceSensitivity": "css",
  "insertPragma": false,
  "jsxBracketSameLine": false,
  "jsxSingleQuote": false,
  "printWidth": 120,
  "proseWrap": "always",
  "quoteProps": "as-needed",
  "requirePragma": false,
  "semi": true,
  "singleQuote": false,
  "tabWidth": 2,
  "trailingComma": "es5",
  "useTabs": false,
  "vueIndentScriptAndStyle": false,
  "filepath": "/Users/martin/dev/src/repro-vscode-prettier-semicolon/test_case.js",
  "parser": "babel",
  "rangeEnd": 795,
  "rangeStart": 39
}
["INFO" - 4:05:39 PM] Formatting completed in 0.008ms.
```

**Comment by darzu** [comment]
I'm pretty suspicious of this commit:
https://github.com/prettier/prettier-vscode/commit/be790ca13b85b26196340ce2a89a1f8a47545605

Indeed, if I change my "on save" mode to "whole file" instead of "modifications", everything works again, no extra semi colon.

**Comment by rotu** [comment]
> I'm pretty suspicious of this commit: [be790ca](https://github.com/prettier/prettier-vscode/commit/be790ca13b85b26196340ce2a89a1f8a47545605)
> 
> Indeed, if I change my "on save" mode to "whole file" instead of "modifications", everything works again, no extra semi colon.

So what that commit does is reports to VS Code just the changed portion of the file, but it shouldn’t cause this sort of problem.

i’m guessing the issue is this or similar: https://github.com/prettier/prettier/issues/12540

**Comment by devinm-hrbl** [comment]
Is this still an issue? I've had to disable prettier entirely because it's adding needless semicolons to my work and causing the compiler to break on every save.

**Comment by rotu** [comment]
It's a bug in Prettier itself. Here it is [reproduced in the Prettier playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEALOAbABzgCc1MBeTYuARwFcBLagCgB0QA6Ael0JLXYBKANysoYrgCpJYzJMwABGAE8iAEzgAzTMADyAIwBWcMDAC+mAAp002SwENiMBmAYF7z6JeIQA5tTQ0ACVaOjgMWXkFAh8iJ2UdDGIGKF8LNBNqGEiuMQlpSKsIDDJ7THdAuDVMDA8bTBgITAA3BDUIYgBBTE7LAElMTQ6G7AYyVqh24gAaTEg6UjhMe0nMGxIcsTgADwIOmDQOCanOgGVwtAYve0rqyhvlKDBMZk19QQoAPh1ZOfQsY4dTqdAgMCh-KCaBi+AD8HAQzThgK6AH17KCUQs8KIoGYcfkZFA5EUSstyjcMtVajB6o0Wm0OgAhHr9QbDGCjcYMmZ-BYZZardbETZQHZ7JyHZGM86BK5QACqBDUHiW9zQj2er30s0gkOhswysugfTUszoSpVahll2gszw9gAXspTmAEI4rnbHcp5RliB9yN9gL8GNpmABCc3K+BWi5yuEpeD+DxytBI7mMkLi5ypE0fLILKDCTBcLiYEI04hEkM1WPQTBjTBQCBYMpSmtG8SdzD10Nh3VQ2HwqCIo7pxk3ODy4h4TAAH1nmD70AHcIRacmTPOYCyc4XS71g7Xo43xEZDicouIuZ+RO7Ix8AHdG3An7w4ocAKKpFJwD-EHzEGwIAALJjJcqQQgOCzJnWQzEPSJ6MkIOLdmYYh4iA0wgBABCeFAaDIKAjiPueCAESg9h4A+9jKARWH6MQ9hgAA1nAMCnPYAC2cAADI-sgmiURk9GMSxbGnO4ripMgMDEGEWFwJx+hVBoag8Ssvh0PYvhwAAYh0nEeNmvjICA9h0I0mEgNgMCcXgADqozwGgklwOc+EMM4zSecoplgIEVkpL6MDeNphkCUJcBYYYaDbKcKS+HgcAAIp0M2cARXgwkgO4izEKZ+j2MpeBWTECb2QwagcsgAAcAAMWExMUcD2YxBCmTE4QkK0Vn0Ol3g4eRZloAAtKKVRVFZ1D0EwcChb44VIIJWVRSAGScQwMlyat4GJSlaXwJl2UwEVFVVdgyAAExYbJ9gMHgCUAMIQJxi0gOEACsVnrAAKkV5HLdlzRhH0kwIOx25uDAnSTKcKhJUdq2MakbknU4yAfXVACMWHIzpX5qMgABsWOXWYZhAA).

**Comment by rotu** [comment]
This is NOT fixed by https://github.com/prettier/prettier/pull/12563.
I opened a new issue here: https://github.com/prettier/prettier/issues/12964

**Comment by medmin** [comment]
any news ? This is annoying.

**Comment by supramaxis** [comment]
one thing I noted is, when i remove return from this code
`                {aniList.map(aniList => {
                  return <Search aniList={aniList} key={aniList.id} />
                })}`

it adds a semicolon like this

`                {aniList.map(aniList => {
                  ;<Search aniList={aniList} key={aniList.id} />
                })}`

I don't know if this is intended or a bug

**Comment by robertocinetto** [comment]
> one thing I noted is, when i remove return from this code ` {aniList.map(aniList => { return <Search aniList={aniList} key={aniList.id} /> })}`
> 
> it adds a semicolon like this
> 
> ` {aniList.map(aniList => { ;<Search aniList={aniList} key={aniList.id} /> })}`
> 
> I don't know if this is intended or a bug

I confirm I see the same behavior. Adding the return seems to prevent the semicolon from being added but it ruins the code cleaning purpose of the syntax

**Comment by jmadsenccri** [comment]
I am still seeing this issue as well. Adds a semicolon after every save to the end of my react component.
```
export const MyComponent = () => {
    return (
        <div>
        ...
        </div>
    );
};;;;;;
```

**Comment by jmadsenccri** [comment]
Worth mentioning that I have `"editor.formatOnSaveMode": "modifications"` and the line with the semicolons being added has not been modified.

**Comment by rotu** [comment]
@jmadsenccri, @robertocinetto see if you can reproduce it in the [Prettier Playground](https://prettier.io/playground/) using the range option.

**Comment by robertocinetto** [comment]
@rotu yes I can replicate it. Please check this in this [Prettier Playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAEnADYkQDqEATiQCYAUwA5tXHLlM-gLz4A6IABalyggDT5W7eHV74A5IKq06ghZIDOOEghjzUAQxKa4k6AFkI6UwHkAbnGriAvgEp8wflG-58OVAYAQmkOHC43YDYYdGoofCh0MhcAbm9fPz8Aeiz8ABVbABFbJHxC6AV9Gzh8akMoOggAW394tganDN19RJa+C0MYIQA6VHIafAYBoeG6huaGDwAqfABGAFEAagB2N2GYCABlGGpw5kXZuAAHEkMwOAYs-n5h57pNrJxmSUFBN3T4rUOLF4gAeOg4ez4MC3TSaAByhiacB4CmIZEoNHoCnwuBguh4wAABgBNaz4QxsfD2HDaA7UBLoJoAIyc+AAJMBGS0XESXPhLNY7I5qITBdUHE4XAA+AGZfCgzQnaDMaVc0KcZjDTQkHD3Bj4AAMklW+D2BwAqlcrk4AMKGUyLfCbKRsMJcbW6-Wrc0QAAyEAA7naHQ8PC5QVkldQVbLAX4WG7NcMEHRNBQcEMGBI-vgAPwCECF-ClRVXer4JUATwJwGAkHI1FKSn40irChcMokhcjmnLUGl+BcGT8oLgTTj-Bg6qTcDkw58MEj47j8q5DG0ulg-3jmQLgleIBHmVKgiCggXx8jEPs0rSPigl6gOCaVxo+gASnA7jBJFzqusqCoHA2CSNUxyDDU-KoDGLQKGwP5KM+r7vooTThAEhgANZOFk5BgMYcCaFkCBIS+b7UPo6HPkYOEMjBzRoRhtFOGRKGUfgxw0FW+AMXBwxZFx1DtgCcAAB4UfodBwEYST6BgWC4AQQk4ERTpeICkBQEq+AANpKtxFqmNQmhaBwQlVkZTiaAAuvI4EwJBDC6TZO4ZABQEgTADBOjwg4afKWk6QZwlWSZ8jUZhdHDJSdRVgwABMxr+Lwg5MMefjVHESJwKUkUsdQwxZVAOVMPg+GEc2pFDr6AbBtQ9qOm44gZf4TTMKURJCDAMBXJoSA5DgwxXHU9iDJSwxgGAWSrAArIaeYvswPCctMIz3DoUyDCMcyNE0TorNshpuLyLW7n4OB0KUOBnfK7g7vKpgwBZYWaBu9KWcZmgPZkQUQLowzkOcIWfdZP0uJILluYC0QgpMV43tCsIIjlPCCGM4mVuW9wALRiTjCX4FcOMAGz4MyzA44GQiZjUTQwDjAAc5M0NJ1A46whhVgThqGiz1BswKIpjEGeM45oYAxmQtTWB0dDiy0EtSyQzKUjjUDQHAgirvKngg69wxNIYVwMFlqWeK1o4WZbmQ4VWhJZcMl3DjANuZcZJXIg7xlFR7OULrrutLd7ThO+1AeB-gWQ63dJ1XlkN46zuLggOIIAQFcSnacgoCxUGAAKlIIJoyAgMYgZcyXafMnUYA4c9OV+uEcDIEYJhmCANd3PXhzY2cyAnOgHfjqydDSXQfr1Mw6CGMwcAAGI0EbPX9yghjoAcqfCDATQkBQNPwH2dxwIcxeZpCmZVqXYBwlv4TGTA+d1MwRut8YphpwAVpoYmHGcugAEV0AQHgG-duadywmScKXVWrISBbxGuEGAGY6BDGQIzY0IARoQFMBQOoVxS4jSIk4RwW8ACOwD4BPwziXNemh1bsHHlvNgFCcBsCfrPV+SA24fxAKYdCYDeHaC4IAyhLduHvw7o5ZkKC0FIASmnE4hgdBnFtM0LhIAiKzS3tUPIhhmS0J4R3ewQ8ACSHRYCHEljgTOABBBoxwaziKMZ2IAA)

The issue happens with the --no-semi option turned on.

**Comment by akshgods** [comment]
same issue

**Comment by itwondersteam** [comment]
anyway to update the prettier used to latest code? main repo is updated with the fix.

**Comment by rotu** [comment]
> anyway to update the prettier used to latest code? main repo is updated with the fix.

I think `npm install -g prettier@next` should do it. You may also need to update the path in vscode settings.

**Comment by appsilva** [comment]
I'm getting this issue now and then (haven't figured out the pattern yet), when I comment code inside the JSX block. As soon as I remove the commented code, it stops adding semicolons.

**Comment by demetriad** [comment]
Very, very annoying. Changed `"editor.formatOnSaveMode` to `file`.

**Comment by VRedondo** [comment]
Why is this closed?

**Comment by demetriad** [comment]
Why is this closed?

**Comment by rotu** [comment]
Now that it’s fixed in prettier v3, it’s waiting on this: https://github.com/prettier/prettier-vscode/issues/2762
